strawberry/experimental/pydantic/utils.py

```diff
@@ -36,3 +43,61 @@ def get_private_fields(cls: Type) -> List[dataclasses.Field]:
             private_fields.append(field)
 
     return private_fields
+
+
+@dataclasses.dataclass()
+class DataclassCreationFields:
+    """Fields required for the fields parameter of make_dataclass"""
+
+    name: str
+    type_annotation: Any
+    field: dataclasses.Field
+
+    @property
+    def to_tuple(self) -> Tuple[str, Any, dataclasses.Field]:
+        # fields parameter wants (name, type, Field)
+        return (self.name, self.type_annotation, self.field)
```

Comment:
could we use a named tuple for this? 😊

https://docs.python.org/3/library/typing.html#typing.NamedTuple

Reply:
sure, i wasn't sure which one strawberry would use internally.